tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+'''
+Utility module to read and write Prometheus Alert rules to file.
+Ensures we can test the Alert rules deployed and for documentation
+purposes.
+'''
+
+import argparse
+import functools
+import ipaddress
+import json
+import operator
+import requests
+import subprocess
+import sys
+import yaml
+
+
+class AlertRuleFile:
+    """A class for manipulating Prometheus Alert & Recording Rules."""
+
+    def __init__(self, ip, port):
+        self.ip = ip
+        self.port = port
+
+    def query_prometheus_api(self, route=False):
+        """Read the Prometheus route contents into memory."""
+        if not route:
+            raise Exception("Expected endpoint route but got None")
+
+        endpoint = "https://{}:{}/api/prometheus/api/v1/{}".format(
+            self.ip, self.port, route
+        )
+        try:
+            response = requests.get(url=endpoint, verify=False)
+        except requests.exceptions.ConnectionError as exc:
+            raise Exception("Failed to reach endpoint {}: {!s}".format(
+                route, exc
+                )
+            )
+        if response.status_code != 200:
+            raise Exception("Expected status code <200> but got {}".format(
+                response.status_code
+                )
+            )
+
+        return response.json()
+
+    def write_to_file(self, content, rule_name):
+        with open('{}.json'.format(rule_name), 'w', encoding='utf-8') as f:
+            f.write(json.dumps(content, ensure_ascii=True))
+
+    def get_dict_element(self, data, path, delimiter='.'):
+        return functools.reduce(operator.getitem, path.split(delimiter), data)
+
+
+def main():
+    """
+        python rule_extractor --ip 127.0.0.1 --port 9090 --type rules
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i',
+        '--ip',
+        dest='ip',
+        help='Specify the target control-plane ingress IP address'
+    )
+    parser.add_argument(
+        '-p',
+        '--port',
+        dest='port',
+        default='8443',
+        help='Specify the target control-plane ingress port'
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='rule_type',
+        help='Specify the rule type. For example: rules or alerts'
+    )
```

Comment:
Use `choices=[]` to provide a list of possible values for this option, and you should rename this `dest` and `help` as it does not really reflect what it is used for:
```suggestion
    parser.add_argument(
        '-t',
        '--type',
        required=True,
        dest='object_type',
        help='Specify the type of objects to fetch',
        choices=['rules', 'alerts']
    )
```
(not sure for the `objects` name, if you find a more relevant naming...)